docs/proposed/grid-manager/managed-grid.rst

```diff
@@ -0,0 +1,338 @@
+(This document is "in-progress", with feedback and input from two
+devchats with Brain Warner and exarkun as well as other input,
+discussion and edits from exarkun. It is NOT done). Search for
+"DECIDE" for open questions.
+
+
+Managed Grid
+============
+
+In a grid using an Introducer, a client will use any storage-server
+the Introducer announces. This means that anyone with the Introducer
+fURL can connect storage to the grid.
+
+Sometimes, this is just what you want!
+
+For some use-cases, though, you want to have clients only use certain
+servers. One case might be a "managed" grid, where some entity runs
+the grid; clients of this grid don't want their uploads to go to
+"unmanaged" storage if some other client decides to provide storage.
+
+One way to limit which storage servers a client connects to is via the
+"server list" (:ref:`server_list`) (aka "Introducer-less"
+mode). Clients are given static lists of storage-servers, and connect
+only to those. This means manually updating these lists if the storage
+servers change, however.
+
+Another method is for clients to use `[client] peers.preferred=`
+configuration option (XXX link? appears undocumented), which suffers
+from a similar disadvantage.
+
+
+Grid Manager
+------------
+
+A "grid-manager" consists of some data defining a keypair (along with
+some other details) and Tahoe sub-commands to manipulate the data and
+produce certificates to give to storage-servers. Certificates assert
+the statement: "Grid Manager X suggests you use storage-server Y to
+upload shares to" (X and Y are public-keys). Such a certificate
+consists of:
+
+ - a version (currently 1)
+ - the public-key of a storage-server
+ - an expiry timestamp
+ - a signature of the above
+
+A client will always use any storage-server for downloads (expired
+certificate, or no certificate) because clients check the ciphertext
+and re-assembled plaintext against the keys in the capability;
+"grid-manager" certificates only control uploads.
+
+
+Grid Manager Data Storage
+-------------------------
+
+The data defining the grid-manager is stored in an arbitrary
+directory, which you indicate with the ``--config`` option (in the
+future, we may add the ability to store the data directly in a grid,
+at which time you may be able to pass a directory-capability to this
+option).
+
+If you don't want to store the configuration on disk at all, you may
+use ``--config -`` (that's a dash) and write a valid JSON
+configuration to stdin.
+
+All commands take the ``--config`` option, and they all behave
+similarly for "data from stdin" versus "data from disk".
+
+
+tahoe grid-manager create
+`````````````````````````
+
+Create a new grid-manager.
+
+If you specify ``--config -`` then a new grid-manager configuration is
+written to stdout. Otherwise, a new grid-manager is created in the
+directory specified by the ``--config`` option. It is an error if the
+directory already exists.
```

Comment:
The reason it's a directory is because the on-disk variant remembers certificates you have issued (with `tahoe grid-manager sign`) whereas the "ephemeral" version doesn't -- so you get to remember those yourself. The entire "grid manager config" is always a single JSON object (in `<dir>/config.json` or as the stdin/out thing).

Perhaps this needs more words to explain, or it's a bad idea entirely. That is, maybe the certificates should be remembered (or not) consistently between the two variants? (The reason the "stdin/out" version *doesn't* do this is because it needs to spit out the actual new certificate when you run `tahoe grid-manager sign`). This was my idea of "making things easier" for the ephemeral version but maybe it's not true. For example, if the new certificate just appeared inside the JSON blob somewhere, you'd have to parse it and pull it out instead of just grabbing all stdout...

Anyway, you're in a much better position to judge which is better since the intended use-case of the "ephemeral" variant is for Ops type use-cases so I'll defer to whichever you think is best. My supposition here is that you'd "create a grid-manager config" and then use it to issue certificates for clients and that workflow would be easier if the "give me a cert" part (`tahoe grid-manager sign`) just spat out the certificate itself.